examples/experimental/dagster-airlift/dagster_airlift/core/airflow_cacheable_assets_def.py

```diff
@@ -0,0 +1,421 @@
+)
+from dagster._record import record
+from dagster._serdes import deserialize_value, serialize_value, whitelist_for_serdes
+from dagster._serdes.serdes import (
+    FieldSerializer,
+    JsonSerializableValue,
+    PackableValue,
+    SerializableNonScalarKeyMapping,
+    UnpackContext,
+    WhitelistMap,
+    pack_value,
+    unpack_value,
+)
+
+from dagster_airlift.core.migration_state import AirflowMigrationState
+
+from .airflow_instance import AirflowInstance, DagInfo, TaskInfo
+from .utils import get_dag_id_from_asset, get_task_id_from_asset
+
+
+class CacheableSpecMappingSerializer(FieldSerializer):
+    def pack(
+        self,
+        mapping: Mapping[str, "CacheableAssetSpec"],
+        whitelist_map: WhitelistMap,
+        descent_path: str,
+    ) -> JsonSerializableValue:
+        return pack_value(SerializableNonScalarKeyMapping(mapping), whitelist_map, descent_path)
+
+    def unpack(
+        self,
+        unpacked_value: JsonSerializableValue,
+        whitelist_map: WhitelistMap,
+        context: UnpackContext,
+    ) -> PackableValue:
+        return unpack_value(unpacked_value, dict, whitelist_map, context)
+
+
+@whitelist_for_serdes(
+    field_serializers={
+        "task_asset_specs": CacheableSpecMappingSerializer,
+        "dag_asset_specs": CacheableSpecMappingSerializer,
+    }
+)
+@record
+class CacheableSpecs:
+    task_asset_specs: Dict[AssetKey, "CacheableAssetSpec"]
+    dag_asset_specs: Dict[AssetKey, "CacheableAssetSpec"]
+
+
+@whitelist_for_serdes
+@record
+class CacheableAssetSpec:
+    # A dumbed down version of AssetSpec that can be serialized easily to and from a dictionary.
+    asset_key: AssetKey
+    description: Optional[str]
+    metadata: Mapping[str, Any]
+    tags: Mapping[str, str]
+    deps: Optional[Sequence["CacheableAssetDep"]]
+    group_name: Optional[str]
+
+    def to_asset_spec(self) -> AssetSpec:
+        return AssetSpec(
+            key=self.asset_key,
+            description=self.description,
+            metadata=self.metadata,
+            tags=self.tags,
+            deps=[AssetDep(asset=dep.asset_key) for dep in self.deps] if self.deps else [],
+            group_name=self.group_name,
+        )
+
+    @staticmethod
+    def from_asset_spec(asset_spec: AssetSpec) -> "CacheableAssetSpec":
+        return CacheableAssetSpec(
+            asset_key=asset_spec.key,
+            description=asset_spec.description,
+            metadata=asset_spec.metadata,
+            tags=asset_spec.tags,
+            deps=[CacheableAssetDep.from_asset_dep(dep) for dep in asset_spec.deps],
+            group_name=asset_spec.group_name,
+        )
+
+
+@whitelist_for_serdes
+@record
+class CacheableAssetDep:
+    # A dumbed down version of AssetDep that can be serialized easily to and from a dictionary.
+    asset_key: AssetKey
+
+    @staticmethod
+    def from_asset_dep(asset_dep: AssetDep) -> "CacheableAssetDep":
+        return CacheableAssetDep(asset_key=asset_dep.asset_key)
```

Comment:
can you create a task for partitioning support here? Would like to do a fast follow and not have to support backwards compat

Reply:
https://linear.app/dagster-labs/issue/FOU-329/cacheable-airflow-assets-partitioning-support